### ISSUE TITLE
File and line freezes (sourceMap problem)

### ISSUE THREAD

**Issue by vinimaz** [dbfannin/ngx-logger#166]
It is not a duplicated from #129, since it is not about the minified main.js and it is the same problem from #154, but it was closed.

The first time NGXLogger prints something in the console it works, but then it freezes the sourceMap information and all the following logs are shown as coming from the same file and line.

I have already turned the sourceMaps on in the angular.json
`"sourceMap": {
              "scripts": true,
              "styles": true,
              "vendor": true
}`

The first log works:

`2020-01-28T09:11:42.314Z INFO [./src/app/app.component.ts:60] Language changed to en`

But then all the following ones show the same line and file

`2020-01-28T09:11:46.752Z DEBUG [./src/app/app.component.ts:60] queryParams event triggered`

The console.log from the browser works correctly.

Is there any solution for it? 
If I use `ng serve --evalSourceMap=true` it works normally, but then the code cannot be debugged and the function is already deprecated.

**Comment by vinimaz** [comment]
I've found out that if I clean the `logPositionRequests `map inside the `mapper.service.ts` everytime, it works.

`private _getSourceMap(sourceMapLocation: string, distPosition: LogPosition): Observable<LogPosition> {
    const req = new HttpRequest<SourceMap>('GET', sourceMapLocation);
// AT THIS IF, if we clean the map and it goes inside, it writes the caller correctly
    if (!this.logPositionRequests.has(sourceMapLocation)) {
      const logPosition = this.httpBackend.handle(req).pipe(
        filter(e => (e instanceof HttpResponse)),
        map<HttpResponse<SourceMap>, SourceMap>((httpResponse: HttpResponse<SourceMap>) => httpResponse.body),
        map<SourceMap, LogPosition>(sourceMap => {
          // map generated position to source position
          return NGXMapperService.getMapping(sourceMap, distPosition);
        }),
        retry(3),
        // if there is an error getting the source, map fall back to the filename and line number of
        catchError(() => {
          return of(distPosition);
        }),
        shareReplay(1)
      );
      this.logPositionRequests.set(sourceMapLocation, logPosition);
    }

    return this.logPositionRequests.get(sourceMapLocation);
  }`

**Comment by dlmoffett** [comment]
Any update on this? I originally found and posted in this duplicate issue: https://github.com/dbfannin/ngx-logger/issues/167

**Comment by yzhan407** [comment]
> I've found out that if I clean the `logPositionRequests `map inside the `mapper.service.ts` everytime, it works.
> 
> at `_getSourceMap()`
> if we clean the map, this condition pass through and it works
> `if (!this.logPositionRequests.has(sourceMapLocation))`

Thanks man!!

I followed your step and built the library locally and it seems to be working!

Here is what I did. I download the code from github. Add one line at mapper.service.ts line 152 which clears the map every time before using it.
![image](https://user-images.githubusercontent.com/23008677/75895861-36c54700-5e04-11ea-8581-5847cce2b5e5.png)
using "ng build" to build the library locally. After building it in the dist/ngx-logger use "npm pack". It generates a tar file. Then I go back to my app, uninstall ngx-logger and install this locally built logger. 
To test the logger I set two loggers at the same file.
![image](https://user-images.githubusercontent.com/23008677/75896832-a5ef6b00-5e05-11ea-8bdd-d38170724c11.png)
Here is what I got in browser console:
![image](https://user-images.githubusercontent.com/23008677/75897005-e2bb6200-5e05-11ea-8ac8-c7d7515703d1.png)
Also I set up a simple fake backend api locally listening to the logger post request. Here is the backend log:
![image](https://user-images.githubusercontent.com/23008677/75897152-172f1e00-5e06-11ea-9451-aedca376b030.png)

Not sure if this is the right way to do it either. But it seems to solve my problem for now at least. If I have more problem I will post it.
